tests/e2e/fullsync_test_for_block_volume.go

```diff
@@ -901,3 +918,219 @@ var _ bool = ginkgo.Describe("full-sync-test", func() {
+			verifySvcPvc := false
+			verifySvcPv := false
+			if entityMetadata.EntityName == svcPVCName {
+				pvc = svcPVC
+				pv = svcPV
+				verifySvcPvc = true
+				verifySvcPv = true
+			} else {
+				pvc = gcPvc
+				pv = gcPv
+			}
+			if verifyPvcEntry || verifySvcPvc {
+				if entityMetadata.EntityName != pvc.Name {
+					framework.Logf("PVC name '%v' does not match PVC name in metadata '%v', for volume id %v",
+						pvc.Name, entityMetadata.EntityName, volumeID)
+					break
+				}
+				if verifyPvEntry || verifySvcPv {
+					if entityMetadata.ReferredEntity == nil {
+						framework.Logf("Missing ReferredEntity in PVC entry for volume id %v", volumeID)
+						break
+					}
+					if entityMetadata.ReferredEntity[0].EntityName != pv.Name {
+						framework.Logf("PV name '%v' in referred entity does not match PV name '%v', "+
+							"in PVC metadata for volume id %v", entityMetadata.ReferredEntity[0].EntityName,
+							pv.Name, volumeID)
+						break
+					}
+				}
+				if pvc.Labels == nil {
+					if entityMetadata.Labels != nil {
+						framework.Logf("PVC labels '%v' does not match PVC labels in metadata '%v', for volume id %v",
+							pvc.Labels, entityMetadata.Labels, volumeID)
+						break
+					}
+				} else {
+					labels := getLabelMap(entityMetadata.Labels)
+					if !(reflect.DeepEqual(labels, pvc.Labels)) {
+						framework.Logf(
+							"Labels on pvc '%v' are not matching with labels in metadata '%v' for volume id %v",
+							pvc.Labels, entityMetadata.Labels, volumeID)
+						break
+					}
+				}
+				if entityMetadata.Namespace != pvc.Namespace {
+					framework.Logf(
+						"PVC namespace '%v' does not match PVC namespace in pvc metadata '%v', for volume id %v",
+						pvc.Namespace, entityMetadata.Namespace, volumeID)
+					break
+				}
+			}
+
+			if verifySvcPvc {
+				svcPVCVerified = true
+			} else {
+				gcPVCVerified = true
+			}
+			continue
+		}
+		if entityMetadata.EntityType == string(cnstypes.CnsKubernetesEntityTypePV) {
+			verifySvcPv := false
+			if entityMetadata.EntityName == svcPV.Name {
+				pvc = svcPVC
+				pv = svcPV
+				verifySvcPv = true
+			} else {
+				pvc = gcPvc
+				pv = gcPv
+			}
+			if verifyPvEntry || verifySvcPv {
+				if entityMetadata.EntityName != pv.Name {
+					framework.Logf("PV name '%v' does not match PV name in metadata '%v', for volume id %v",
+						pv.Name, entityMetadata.EntityName, volumeID)
+					break
+				}
+				if pv.Labels == nil {
+					if entityMetadata.Labels != nil {
+						framework.Logf("PV labels '%v' does not match PV labels in metadata '%v', for volume id %v",
+							pv.Labels, entityMetadata.Labels, volumeID)
+						break
+					}
+				} else {
+					labels := getLabelMap(entityMetadata.Labels)
+					if !(reflect.DeepEqual(labels, pv.Labels)) {
+						framework.Logf(
+							"Labels on pv '%v' are not matching with labels in pv metadata '%v' for volume id %v",
+							entityMetadata.Labels, pv.Labels, volumeID)
```

Comment:
entityMetadata.Labels, pv.Labels, to be 
pv.Labels, entityMetadata.Labels,

Reply:
thanks, fixed it